### PR TITLE
Rename `gen-key` and improve docs

### PR DESCRIPTION
**Pull request by outkine**
Improve README.md with examples and a clarification about how to pass
command line arguments

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,13 +67,13 @@ DREAM_INTERFACE env)
...
     tokens. By default, a random secret is generated on each call to
-    Dream.run. For production, generate a key with gen-key and load it
+    Dream.run. For production, generate a key with gen-secret and load it
     from file. A medium-sized Web app serving 1000 fresh encrypted
     cookies per second should rotate keys about once a year. See
     argument --old-secrets) below for key rotation.
 ```
 
-Or as environmnet variables:
+Or as environmental variables:
```
Comment (aantron) [inline README.md]: AFAIK this should remain "environment variables."

File: README.md [hunk 2]
```diff
@@ -113,11 +113,27 @@ DREAM_VERBOSITY
     See option --verbosity.
 ```
 
+Note that in order for the command-line arguments to be recognized by Dream, and not by the build tool, it is necessary to prepend them with a double dash:
+
+```sh
+# esy
+npx esy start -- -p 9000
+# dune
+dune exec ./main.ml -- -p 9000
```
Comment (tmattio) [inline README.md]: I don't think `esy start` exists, and `dune exec` should be called with the name of the executable, or the path with the `.exe` extension.
```suggestion
# esy
npx esy run <my_binary> -- -p 9000
# dune
dune exec <my_binary> -- -p 9000
```

File: README.md [hunk 3]
```diff
@@ -113,11 +113,27 @@ DREAM_VERBOSITY
...
 The following subcommands are provided in the generated CLI for convenience:
 
-- `gen-key` - Generate a random secret key.
+- `gen-secret` - Generate a random secret.
+
+```sh
+# esy
+npx esy start gen-secret
+# dune
+dune exec ./main.ml gen-secret
```
Comment (tmattio) [inline README.md]: Same as above:
```suggestion
# esy
npx esy run <my_binary> gen-secret
# dune
dune exec <my_binary> gen-secret
```

**Review by tmattio — COMMENTED** [review]
Thanks @outkine, I left a few comments for the example commands, once this is addressed, the PR is good to be merged 🙂

File: README.md (hunk 2 above)
Comment (outkine) [reply README.md:122]: Ah you're right, I guess `start` is a project-specific command

**Comment by outkine** [comment]
Closes #2

**Pull request merged**